lntest/itest/lnd_test.go

```diff
@@ -5171,6 +5171,137 @@ func testListChannels(net *lntest.NetworkHarness, t *harnessTest) {
+	chanPoint := openChannelAndAssert(
+		ctxt, t, net, alice, bob,
+		lntest.OpenChannelParams{
+			Amt: chanAmt,
+		},
+	)
+
+	// Wait for Alice and Bob to receive the channel edge from the
+	// funding manager.
+	ctxt, _ = context.WithTimeout(ctxb, defaultTimeout)
+	err = alice.WaitForNetworkChannelOpen(ctxt, chanPoint)
+	if err != nil {
+		t.Fatalf("alice didn't see the alice->bob channel before "+
+			"timeout: %v", err)
+	}
+
+	ctxt, _ = context.WithTimeout(ctxb, defaultTimeout)
+	err = bob.WaitForNetworkChannelOpen(ctxt, chanPoint)
+	if err != nil {
+		t.Fatalf("bob didn't see the bob->alice channel before "+
+			"timeout: %v", err)
+	}
+
+	assertEdgeDisabled := func(node *lntest.HarnessNode,
+		chanPoint *lnrpc.ChannelPoint, disabled bool) {
+
+		var predErr error
+		err = wait.Predicate(func() bool {
+			req := &lnrpc.ChannelGraphRequest{
+				IncludeUnannounced: true,
+			}
+			ctxt, _ = context.WithTimeout(ctxb, defaultTimeout)
+			chanGraph, err := node.DescribeGraph(ctxt, req)
+			if err != nil {
+				predErr = fmt.Errorf("unable to query node %v's graph: %v", node, err)
+				return false
+			}
+			numEdges := len(chanGraph.Edges)
+			if numEdges != 1 {
+				predErr = fmt.Errorf("expected to find 1 edge in the graph, found %d", numEdges)
+				return false
+			}
+			edge := chanGraph.Edges[0]
+			if edge.ChanPoint != chanPoint.GetFundingTxidStr() {
+				predErr = fmt.Errorf("expected chan_point %v, got %v",
+					chanPoint.GetFundingTxidStr(), edge.ChanPoint)
+			}
+			var policy *lnrpc.RoutingPolicy
+			if node.PubKeyStr == edge.Node1Pub {
+				policy = edge.Node1Policy
+			} else {
+				policy = edge.Node2Policy
+			}
+			if disabled != policy.Disabled {
+				predErr = fmt.Errorf("expected policy.Disabled to be %v, "+
+					"but policy was %v", disabled, policy)
+				return false
+			}
+			return true
+		}, defaultTimeout)
+		if err != nil {
+			t.Fatalf("%v", predErr)
+		}
+	}
+
+	sendReq := func(node *lntest.HarnessNode, chanPoint *lnrpc.ChannelPoint,
+		action routerrpc.SetChannelStatusAction) {
+
+		req := &routerrpc.SetChannelStatusRequest{
+			ChanPoint: chanPoint,
+			Action:    action,
+		}
+		ctxt, _ = context.WithTimeout(ctxb, defaultTimeout)
+		_, err = node.RouterClient.SetChannelStatus(ctxt, req)
+		if err != nil {
+			t.Fatalf("unable to call SetChannelStatus for %s's node: %v",
+				node.Name(), err)
+		}
+	}
+
+	// The channel should not be disabled initially.
+	assertEdgeDisabled(alice, chanPoint, false)
+
+	// Manually disable the channel.
+	sendReq(alice, chanPoint, routerrpc.SetChannelStatusAction_DISABLE)
+	assertEdgeDisabled(alice, chanPoint, true)
+
+	// Manually re-enable the channel.
+	sendReq(alice, chanPoint, routerrpc.SetChannelStatusAction_ENABLE)
+	assertEdgeDisabled(alice, chanPoint, false)
+
+	// Restoring automatic channel state management should be a no-op.
+	sendReq(alice, chanPoint, routerrpc.SetChannelStatusAction_AUTO)
+	assertEdgeDisabled(alice, chanPoint, false)
+}
```

Comment:
We can test the switch to auto management by closing out the channel here an asserting a disable message is sent out. You can likely lift/consolidate some of the code here from a related test that asserts the same behavior: https://github.com/lightningnetwork/lnd/blob/master/lntest/itest/lnd_test.go#L13283

Reply:
A disable message will be sent out on channel close (or peer disconnect) whether or not we switched back to auto management, since we don't want to clutter the network with falsely advertised enabled channels.

However, I did add more tests for auto management:
* after manual disable: reconnecting peer shouldn't automatically re-enable channel
* after auto management restored: reconnecting peer SHOULD automatically re-enable channel